File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.cpp
class Solution {
public:
    int minimumPushes(string word) {
        unordered_map<char,int> mp;
        priority_queue<int>pq;
        int n = word.length();
        for(int i=0;i<n;i++) {
            mp[word[i]-'a']++;
        }
        for(auto it=mp.begin();it!=mp.end();it++) {
            pq.push(it->second);
        }
        int tot_push = 0,ind = 0;
        while(!pq.empty()) {
            tot_push +=(1+(ind/8))*pq.top();
            pq.pop();
            ind++;
        }
        return tot_push;
    }
};

File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.java
class Solution {
    public int minimumPushes(String word) {
        int[] arr = new int[26];
        int j=0,cnt=1,ans=0;
        for(int i=0;i<word.length();i++) {
            arr[word.charAt(i)-'a']++;
        }
        Arrays.sort(arr);
        for(int i=25;i>=0;i--) {
            ans+=cnt*arr[i];
            j++;
            if(j%8==0) cnt++;
        }
        return ans;
    }
}